ArduCopter/mode_autorotate.cpp

```diff
@@ -268,34 +370,39 @@ void ModeAutorotate::run()
         }
 
         break;
-        }
+    }
     }
 
+    float pilot_roll, pilot_pitch, pilot_yaw_rate;
+    // Operator is in control of roll and yaw.  Controls act as if in stabilise flight mode.  Pitch
+    // is controlled by speed-height controller.
+    get_pilot_desired_lean_angles(pilot_roll, pilot_pitch, copter.aparm.angle_max, copter.aparm.angle_max);
 
-    switch (nav_pos_switch) {
+    float target_roll;
 
-        case Navigation_Decision::USER_CONTROL_STABILISED:
-        {
-            // Operator is in control of roll and yaw.  Controls act as if in stabilise flight mode.  Pitch 
-            // is controlled by speed-height controller.
-            float pilot_roll, pilot_pitch;
-            get_pilot_desired_lean_angles(pilot_roll, pilot_pitch, copter.aparm.angle_max, copter.aparm.angle_max);
+    // Grab inertial velocity
+    const Vector3f& vel = inertial_nav.get_velocity_neu_cms();
 
-            // Get pilot's desired yaw rate
-            float pilot_yaw_rate = get_pilot_desired_yaw_rate(channel_yaw->norm_input_dz());
+    // rotate roll, pitch input from north facing to vehicle's perspective
+    float roll_vel =  vel.y * ahrs.cos_yaw() - vel.x * ahrs.sin_yaw(); // body roll vel
+    float pitch_vel = vel.y * ahrs.sin_yaw() + vel.x * ahrs.cos_yaw(); // body pitch vel
```

Comment:
You can do this bit using `earth_to_body2D` something like:
```
Vector2f vel = ahrs.earth_to_body2D(inertial_nav.get_velocity_neu_cms().xy());
```
Then you could extract, or keep it in the Vector2:
```
float roll_vel = vel.y;
float pitch_vel = vel.x;
```
Using known good functions makes it easier to review (although this does look correct) and also saves flash.